### ISSUE TITLE
Provide an api to re-link TS groups

### ISSUE THREAD

**Issue by luisiacc** [nvim-treesitter/nvim-treesitter#3192]
**Is your feature request related to a problem? Please describe.**
When changing colorschemes with the `:colorscheme` command, most colorschemes implementations have a `hi clear` somewhere on their code, sometimes they define TS groups, an example: 

- Start neovim with a scheme that defined `TSKeywordReturn`
- Change the colorscheme to something else that doesn't explicitly define `TSKeywordReturn`, but it does define `TSKeyword`
- The `TSKeywordReturn` will be cleared, and therefore not been highlighted

On neovim start, `TSKeywordReturn` is linked to `TSKeyword` which is linked to `Keyword`, so then it's not a problem, but it is when changing colorschemes, because that group won't bee highlighted after changing the colorschemes, and there is no go-to way to re link the TS groups, something like:

```lua
local highlights = require("nvim_treesitter.highlight_groups")

-- the user set the highlights
```

**Describe the solution you'd like**
Highlights on https://github.com/nvim-treesitter/nvim-treesitter/blob/master/plugin/nvim-treesitter.lua to be provided by nvim-treesitter in a module, right now is just a local variable, something like:

```lua
local utils = require("nvim_treesitter.utils")

utils.reset_ts_highlights()
```

**Describe alternatives you've considered**
Maybe you can provide a function that re-links those groups instead of giving the entire `highlights` object, I guess that's easier for the user, doesn't have to manually link the groups themselves

**Additional context**
Most users may not note this changes because not every day people change colorschemes continuosly in the same neovim session, or maybe they do but because the colors are off so they just reset neovim with the new colorschems and then it fixes it so they move on with their day?

Maybe this could be recommended to do on the docs to colorschemes authors to do after they set their colorscheme?, or between they do `hi clear` and setting the colorscheme?

I change colorschemes often on the same neovim session, as a workaround I have copied the `highlights` object you guys have and set the highlights before every colorscheme change, making a `Colorscheme` user command to handle that, as seen on [this config file](https://github.com/luisiacc/nvim-config/blob/main/lua/luisiacc/colorscheme.lua), although now that I think about it, I could just do it on an `autocmd Colorscheme` 🤔.

**Comment by clason** [comment]
Indeed, local additions to color schemes should be handled via the (built-in!) `ColorScheme` autocommand; this is standard Vim practice (see https://gist.github.com/romainl/379904f91fa40533175dfaec4c833f2f) and not specific to treesitter.

**Comment by urxvtcd** [comment]
So I've noticed that there was some work done recently to fix this, as `plugin/nvim-treesitter.lua` contains this snippet:
```lua
api.nvim_create_autocmd("ColorScheme", {
  group = augroup,
  callback = highlight.set_default_hlgroups,
  desc = "Set default highlights",
})
```
by @jdrouhard.
However, this doesn't seem to fix the issue for me. It is fixed by running this, though:
```
source ~/.vim/pack/bundle/opt/nvim-treesitter/lua/nvim-treesitter/highlight.lua
```
Which means that calling `highlight.set_default_hlgroups` is not enough.
